daemon/container.go

```diff
@@ -1557,3 +1548,37 @@ func (c *Container) LogDriverType() string {
 	}
 	return c.hostConfig.LogConfig.Type
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	all, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(all) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := filepath.Dir(basePath)
+	for baseDir != "" && !strings.HasPrefix(all[0], baseDir) {
+		baseDir = filepath.Dir(baseDir)
+	}
+
+	relative := make([]string, len(all))
+
+	p := fmt.Sprintf("%s%s", baseDir, string(filepath.Separator))
+	for i, f := range all {
+		r := strings.Replace(f, p, "", -1)
```

Comment:
I think doing `r := f[len(p):]` would be faster and safer since the `-1` could force the replacement to happen multiple times in for the path, and not just the leading bit.
```
l := len(fmt.Sprintf("%s%s", baseDir,string(filepath.Separator))
for ...
   r := f[l:]
   ...
```